test/cmdlineTests/yul_to_wasm_source_location_crash/output

```diff
@@ -0,0 +1,5190 @@
+
+======= yul_to_wasm_source_location_crash/input.yul (Ewasm) =======
+
+Pretty printed source:
+/// @use-src 0:"test.sol"
+object "C" {
+    code { }
+}
+
+
+==========================
+
+Translated source:
+/// @use-src 0:"test.sol"
+object "C" {
```

Comment:
What a monster :-).

Maybe use

```
/// @use-src 0:"test.sol"
object "C" {
    code { sstore(0,0) }
}
```

as test case instead and run it with
``--strict-assembly --yul-dialect evm --machine ewasm --optimize``

:-)?

That also triggered the bug and is fixed with the PR, but it doesn't require 5K of test expectation ;-).

Reply:
We have #12074 now so you can use output selection to make this smaller. `--ir-optimized` or `--bin` should be enough in this case.

Reply:
i'll test with the new command parameters and see if they also trigger the problem

Reply:
I think they should. For the compiler we actually skip generating the stuff if output is not required but assembler is more straightforward. Output selection just makes it not print some stuff but it still does all the work. Maybe that's actually something we should change though.

Reply:
Both parameters (together and on their own) make it not crash.

Reply:
Adding `sstore(0,0)` makes it crash again though.

Reply:
Then it sounds like it might be because of the optimizer?

Option selection should not disable anything unless some actual work is being done inside functions like `print()` or `toHex()`. https://github.com/ethereum/solidity/blob/v0.8.10/solc/CommandLineInterface.cpp#L1036-L1091

Reply:
I think I misspelled something, I am currently testing again

Reply:
```
crash:
	with sstore(0,0):
		--strict-assembly --yul-dialect evm --machine ewasm
		--strict-assembly --yul-dialect evm --machine ewasm --optimize
	without sstore(0,0):
		--strict-assembly --yul-dialect evm --machine ewasm
No crash:
	with and without sstore(0,0):
		--strict-assembly --yul-dialect evm --machine ewasm --ir-optimized
		--strict-assembly --yul-dialect evm --machine ewasm --ir-optimized --bin
	with no sstore(0,0):
		--strict-assembly --yul-dialect evm --machine ewasm --optimize
```

output wise I guess it's best to go with
```
	with sstore(0,0):

		--strict-assembly --yul-dialect evm --machine ewasm --optimize
```

Reply:
`--strict-assembly --yul-dialect evm --machine ewasm --ewasm-ir` also reproduces the crash for me on 0.8.10 (without `sstore` if `--optimize` is not used).

I guess the thing that triggers the crash is called from within `AssemblyStack::print()` after all.